completers/nix-build_completer/cmd/root.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"github.com/rsteube/carapace"
+	"github.com/rsteube/carapace-bin/pkg/actions/tools/nix"
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "nix-build",
+	Short: "build a Nix expression",
+	Long:  "https://nixos.org/manual/nix/stable/command-ref/nix-build.html",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+func Execute() error {
+	return rootCmd.Execute()
+}
+func init() {
+	carapace.Gen(rootCmd).Standalone()
+
+	rootCmd.Flags().StringP("include", "I", "", "Include paths")
+	rootCmd.Flags().Bool("no-out-link", false, "Do not create a symlink to the output path")
+	rootCmd.Flags().StringP("out-link", "o", "", "Change the name of the symlink to the output path")
+	rootCmd.Flags().Bool("dry-run", false, "Show what store paths would be built or downloaded")
+	rootCmd.Flags().StringP("attr", "A", "", "Attribute to build")
+	// TODO support --arg and --argstr
+
+	carapace.Gen(rootCmd).PositionalAnyCompletion(nix.ActionPaths())
+	carapace.Gen(rootCmd).FlagCompletion(carapace.ActionMap{
+		"attr": nix.ActionAttr(rootCmd, func(_ *cobra.Command, c carapace.Context) string {
+			if len(c.Args) > 0 {
+				return c.Args[0]
```

Comment:
If we use `Flags().String("arg", ...)` for `--arg` and `--argstr` then the "value" component becomes a positional arg but we need to know the actual positional arg to figure out what we're trying to get the arguments out for.

While I can get the `--arg` flags using `os.Args` in `ActionAttr` but I'd still need some way to figure out what the positional argument for the command is.

Reply:
Yes flags taking multiple arguments (nargs) is what has blocked the nix completion so far.
It's a feature i consider a bad idea to be honest.
I'm trying to add support for it though, pflag is already patched and i might get it working with https://github.com/rsteube/carapace/pull/597 .